inst/shiny/modules/slope_selector.R

```diff
@@ -275,73 +343,68 @@ slope_selector_server <- function(
     #' Render manual slopes table
     refresh_reactable <- reactiveVal(1)
     output$manual_slopes <- renderReactable({
-      log_trace("{id}: rendering slope edit data table")
 
-      # enclosing this in isolate() so the table is not re-rendered on every edit #
+      log_trace("{id}: rendering slope edit data table")
+      # Isolate to prevent unnecessary re-renders on every edit
       isolate({
         data <- manual_slopes()
       })
 
+      # Fixed columns (TYPE, RANGE, REASON)
+      fixed_columns <- list(
+        TYPE = colDef(
+          cell = dropdown_extra(
+            id = session$ns("edit_TYPE"),
+            choices = c("Selection", "Exclusion"),
+            class = "dropdown-extra"
+          ),
+          width = 200
+        ),
+        RANGE = colDef(
+          cell = text_extra(
+            id = session$ns("edit_RANGE")
+          )
+        ),
+        REASON = colDef(
+          cell = text_extra(
+            id = session$ns("edit_REASON")
+          ),
+          width = 400
+        )
+      )
+
+      # Dynamic group column definitions
+      dynamic_columns <- lapply(slopes_groups(), function(col) {
+        colDef(
+          cell = dropdown_extra(
+            id = session$ns(paste0("edit_", col)),
```

Comment:
nitpick: since `ns` is extracted from `session` object at the top of the `moduleServer`:
```suggestion
            id = ns(paste0("edit_", col)),
```